### PR TITLE
SRC-6 example contract does not update managed assets

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Bug fix

## Changes

The following changes have been made:

- There is a critical flaw in the SRC-6 example contract where the managed assets is not updated with assets are withdrawn. This can be exploited to drain the contract. This PR resolves this critical bug.

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.
- [x] I have updated the changelog to reflect the changes on this PR.

### CODE REVIEW

File: examples/src6-vault/multi_asset_vault/src/main.sw [hunk 1]
```diff
@@ -81,6 +81,9 @@ impl SRC6 for Contract {
         require(msg_asset_id() == share_asset_id, "INVALID_ASSET_ID");
         let assets = preview_withdraw(share_asset_id, shares);
 
+        let mut vault_info = storage.vault_info.get(share_asset_id).read();
+        vault_info.managed_assets = vault_info.managed_assets - shares;
```
Comment (SwayStar123) [inline examples/src6-vault/multi_asset_vault/src/main.sw:85]: you need to set the storage here right? AFAIK the mut doesnt make the storage mutate aswell

File: examples/src6-vault/single_asset_vault/src/main.sw [hunk 2]
```diff
@@ -82,6 +82,9 @@ impl SRC6 for Contract {
         require(msg_asset_id() == share_asset_id, "INVALID_ASSET_ID");
         let assets = preview_withdraw(share_asset_id, shares);
 
+        let mut vault_info = storage.vault_info.get(share_asset_id).read();
+        vault_info.managed_assets = vault_info.managed_assets - shares;
```
Comment (SwayStar123) [inline examples/src6-vault/single_asset_vault/src/main.sw:86]: same here

**Review by SwayStar123 — CHANGES_REQUESTED** [review]

File: examples/src6-vault/multi_asset_vault/src/main.sw (hunk 1 above)
Comment (bitzoic) [reply examples/src6-vault/multi_asset_vault/src/main.sw:85]: `mut` is required because a subtraction occurs on line 85.

Comment (SwayStar123) [reply examples/src6-vault/multi_asset_vault/src/main.sw:85]: but does the change reflect in storage?

**Review by SwayStar123 — DISMISSED** [review]

**Review by K1-R1 — APPROVED** [review]

**Review by SwayStar123 — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**